tests/vector_stores/test_weaviate.py

```diff
@@ -0,0 +1,30 @@
+import sys
+from unittest.mock import MagicMock
+from gpt_index.data_structs.node_v2 import Node
+from gpt_index.vector_stores.types import NodeEmbeddingResult
+
+from gpt_index.vector_stores.weaviate import WeaviateVectorStore
+
+
+def test_weaviate_add() -> None:
+    # mock import
+    sys.modules["weaviate"] = MagicMock()
+    weaviate_client = MagicMock()
+    batch_context_manager = MagicMock()
+    weaviate_client.batch.__enter__.return_value = batch_context_manager
+
+    vector_store = WeaviateVectorStore(weaviate_client=weaviate_client)
+
+    vector_store.add(
+        [
+            NodeEmbeddingResult(
+                id="test node id",
+                node=Node(text="test node text"),
+                embedding=[0.5, 0.5],
+                doc_id="test doc id",
+            )
+        ]
+    )
+
+    args, _ = batch_context_manager.add_data_object.call_args
```

Comment:
Should we validate more args than just the vector? Up to you 👍

Reply:
Only did vector because I was trying to reproduce potential bug.

I definitely think we need better test coverage in general (and refactoring code to make testing easier to enable that).

Reply:
Not going to do that yet in this PR.